### PR TITLE
fix: use `rand` `v0.9` whenever possible

### PR DESCRIPTION
**Pull request by antbern**
## Motivation and Context

This makes the core functionality (except for storing kdbx files) work under `node` 💯 

## Summary

Provide a short summary of changes in this pull request.

- In places where possible, directly inject the random bytes (eg. entropy) instead of letting each library use their own (old) version of the `rand` crate which does not play nice with `node` at the moment.
- Remove the need to directly depend on the old version of `rand` 💯 

Note: not all use of the old version is fixed. The backup generation using the `kdbx` crate will continue to not work under `node`.

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [x] Code follows project style and guidelines.
- [x] No redundant or duplicate code.
- [x] No added new dependencies without clear justification and approval.

### Documentation

- [x] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [x] Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.
- [x] Tests pass locally.

### Build & CI/CD

- [x] Confirmed no CI/CD pipeline issues.
- [x] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

File: crates/etopay-wallet/src/rebased/crypto/ed25519.rs [hunk 1]
```diff
@@ -146,7 +147,13 @@ impl Ed25519KeyPair {
     }
 
     pub fn generate<R: CryptoRng + RngCore>(rng: &mut R) -> Self {
-        let kp = ed25519_consensus::SigningKey::new(rng);
+        // let kp = ed25519_consensus::SigningKey::new(rng);
```
Comment (kristifidani) [inline crates/etopay-wallet/src/rebased/crypto/ed25519.rs:150]: Remove this ? 😄 
```suggestion
```

**Review by kristifidani — APPROVED** [review]
Noice 💯

File: crates/etopay-wallet/src/rebased/crypto/ed25519.rs (hunk 1 above)
Comment (kristifidani) [reply crates/etopay-wallet/src/rebased/crypto/ed25519.rs]: ah yes then leave it 😄
